rust/main/chains/hyperlane-cosmos/src/mailbox/delivery_indexer.rs

```diff
@@ -123,7 +123,21 @@ impl Indexer<H256> for CosmosMailboxDeliveryIndexer {
     }
 
     async fn get_finalized_block_number(&self) -> ChainResult<u32> {
-        self.indexer.get_finalized_block_number().await
+        self.provider.get_finalized_block_number().await
+    }
+
+    async fn fetch_logs_by_tx_hash(
```

Comment:
fwiw I don't expect these functions atm to be used for indexing deliveries or for message dispatches (just bc it's only meant to be used for indexing hook events), but having them doesn't hurt